File: src/main/java/eu/mikroskeem/moarsms/bukkit/BukkitPlatform.java
package eu.mikroskeem.moarsms.bukkit;

import eu.mikroskeem.moarsms.HTTPServerThread;
import eu.mikroskeem.moarsms.Platform;
import eu.mikroskeem.utils.text.MinecraftText;
import lombok.RequiredArgsConstructor;

import java.util.HashMap;
import java.util.Map;

@RequiredArgsConstructor
public class BukkitPlatform implements Platform {
    private final HTTPServerThread httpServerThread;
    private final plugin pl;
    @Override public void shutdown() {
        if (httpServerThread != null) {
            httpServerThread.interrupt();
        }
    }

    @Override public Map<String, String> getServiceSecrets() {
        return new HashMap<String, String>(){{
            pl.getConfig().getConfigurationSection("services").getValues(false).forEach((cServiceId,val)->{
                put(cServiceId, pl.getConfig().getString(String.format("services.%s.secret", cServiceId)));
            });
        }};
    }

    @Override public String invokeService(String serviceId, String message) {
        String username = message.trim();
        if(!MinecraftText.validateUsername(username)){
            return this.getMessage("badmessage.badUsername");
        }
        pl.getServiceCommands(serviceId).forEach(command->{
            pl.getServer().getScheduler().runTask(pl, ()->pl.runCommand(command.replaceAll("%user%", username)));
        });
        return this.getMessage("success.thanks");
    }

    @Override public boolean allowTest() {
        return pl.getConfig().getBoolean("config.allowTest", false);
    }

    @Override public String getMessage(String path) {
        return pl.getConfig().getString(String.format("messages.%s", path), "");
    }
}

File: src/main/java/eu/mikroskeem/moarsms/bukkit/BukkitPlatform.kt
package eu.mikroskeem.moarsms.bukkit

import eu.mikroskeem.moarsms.Platform

/**
 * @author Mark Vainomaa
 */
internal class BukkitPlatform(private val plugin: MoarSMSPlugin): Platform {
    private val usernamePattern = "^[A-Za-z0-9_]+$".toRegex()

    override val serviceSecrets get() = HashMap<String, String>().apply {
        plugin.config.getConfigurationSection("services").getValues(false).forEach { cServiceId, _ ->
            put(cServiceId, plugin.config.getString("services.$cServiceId.secret"))
        }
    }

    override val allowTest get() = plugin.config.getBoolean("config.allowTest", false)

    override val logger get() = plugin.logger!!

    override val fortumoUtils get() = plugin.fortumoUtils

    override fun getMessage(path: String): String = plugin.config.getString("messages.$path", "")

    override fun invokeService(serviceId: String, message: String): String {
        val username = message.trim { it <= ' ' }
        if(username.length !in 3..16 || !username.matches(usernamePattern)) {
            return this.getMessage("badmessage.badUsername")
        }
        plugin.config.getStringList("services.$serviceId.commands").forEach {
            plugin.server.scheduler.runTask(plugin) {
                plugin.server.dispatchCommand(plugin.server.consoleSender, it.replace("%user%", username))
            }
        }
        return this.getMessage("success.thanks")
    }
}